src/components/DragAndDropExample.js

```diff
@@ -0,0 +1,280 @@
+            this.setState({ itemLinkedZone });
+            break;
+          }
+        }
+        if (!linkedZoneFound) {
+          itemLinkedZone[index] = -1;
+          this.setState({ itemLinkedZone });
+        }
+        Animated.spring(this.state.pan[index], {
+          toValue: { ...springTo }
+        }).start();
+        this.setState({ movingItem: -1 });
+      }
+    });
+  };
+
+  updateZoneLayout = (event, index) => {
+    const layout = event.nativeEvent.layout;
+    this.zoneLayouts[index] = layout;
+    const itemIndex = this.state.itemLinkedZone.indexOf(index);
+    if (itemIndex >= 0) {
+      const springTo = {
+        x: layout.x + layout.width / 2 - this.itemLayouts[itemIndex].width / 2,
+        y: layout.y + layout.height / 2 - this.itemLayouts[itemIndex].height / 2
+      };
+      Animated.spring(this.state.pan[itemIndex], {
+        toValue: { ...springTo }
+      }).start();
+    }
+  };
+
+  renderZones = n => {
+    const zones = [];
+    for (let i = 0; i < n; i++) {
+      const itemIndex = this.state.itemLinkedZone.indexOf(i);
+      zones.push(
+        <View
+          key={`zone_${i}`}
+          onLayout={e => this.updateZoneLayout(e, i)}
+          style={{
+            justifyContent: 'center',
+            alignItems: 'center',
+            width: itemIndex >= 0 ? this.itemLayouts[itemIndex].width : 50,
+            height: 70
+          }}
+        >
+          <View style={styles.dot} />
+        </View>
+      );
+    }
+    return (
+      <View
+        style={{
+          flexDirection: 'row',
+          flexWrap: 'wrap',
+          justifyContent: 'center',
+          backgroundColor: 'orange'
+        }}
+      >
+        {zones}
+      </View>
+    );
+  };
+
+  updateItemLayouts = (event, index) => {
+    const layout = event.nativeEvent.layout;
+    this.itemLayouts[index] = layout;
+    this.initialPositions[index] = { x: layout.x, y: layout.y };
+    this.state.pan[index] = new Animated.ValueXY(this.initialPositions[index]);
+    this.initPanResponder(index);
+    this.setState({ pan: this.state.pan });
+  };
+
+  renderItems = sentence => {
+    // the idea is to place objects first on the screen using html (flex grid
+    // here, for example), and only then make it interactable
+    // and initialize the coordinates
+    const wordArray = sentence.split(' ');
+    const allItemsInitialized = this.state.pan.reduce(
+      (acc, val) => acc && val !== null,
+      true
+    );
+
+    // if all items are initialized -> move container to top
+    // (to use as a relative object for items)
+    let containerStyle = { position: 'absolute', top: 0, left: 0 };
+    if (!allItemsInitialized) {
+      containerStyle = {
+        ...containerStyle,
+        width: '100%',
+        paddingTop: Window.height / 2,
+        justifyContent: 'space-around',
+        alignItems: 'center',
+        flexDirection: 'row',
+        flexWrap: 'wrap'
+      };
+    }
+
+    // initialize items params
```

Comment:
I would do this somewhere else. Not sure if we want to do this in `render` (even if this is done only once). Why not `componentDidMount` / `componentWillMount` or something like that?